uberduck_ml_dev/trainer/gradtts.py

```diff
@@ -0,0 +1,247 @@
+                hifigan_checkpoint="/home/w_uberduck_ai/Speech-Backbones/Grad-TTS/checkpts/gen_02640000_studio",
+                cudnn_enabled=self.hparams.cudnn_enabled,
+            )
+            return audio
+
+    def train(self, checkpoint=None):
+        if self.distributed_run:
+            self.init_distributed()
+
+        train_dataset = TextMelDataset(
+            self.hparams.training_audiopaths_and_text,
+            self.hparams.text_cleaners,
+            1.0,
+            self.hparams.n_feats,
+            self.hparams.sampling_rate,
+            self.hparams.mel_fmin,
+            self.hparams.mel_fmax,
+            self.hparams.filter_length,
+            self.hparams.hop_length,
+            (self.hparams.filter_length - self.hparams.hop_length) // 2,
+            self.hparams.win_length,
+            intersperse_text=self.hparams.intersperse_text,
+            intersperse_token=(len(SYMBOL_SETS[self.hparams.symbol_set])),
+            symbol_set=self.hparams.symbol_set,
+        )
+        collate_fn = TextMelCollate()
+
+        loader = DataLoader(
+            dataset=train_dataset,
+            batch_size=self.hparams.batch_size,
+            collate_fn=collate_fn,
+            drop_last=True,
+            num_workers=0,
+            shuffle=False,
+        )
+
+        test_dataset = TextMelDataset(
+            self.hparams.test_audiopaths_and_text,
+            self.hparams.text_cleaners,
+            1.0,
+            self.hparams.n_feats,
+            self.hparams.sampling_rate,
+            self.hparams.mel_fmin,
+            self.hparams.mel_fmax,
+            self.hparams.filter_length,
+            self.hparams.hop_length,
+            (self.hparams.filter_length - self.hparams.hop_length) // 2,
+            self.hparams.win_length,
+            intersperse_text=self.hparams.intersperse_text,
+            intersperse_token=(len(SYMBOL_SETS[self.hparams.symbol_set])),
+            symbol_set=self.hparams.symbol_set,
+        )
+
+        model = GradTTS(self.hparams)
+
+        if self.hparams.checkpoint:
+            model.load_state_dict(torch.load(self.hparams.checkpoint))
+        model = model.cuda()
+
+        print(
+            "Number of encoder + duration predictor parameters: %.2fm"
+            % (model.encoder.nparams / 1e6)
+        )
+        print("Number of decoder parameters: %.2fm" % (model.decoder.nparams / 1e6))
+        print("Total parameters: %.2fm" % (model.nparams / 1e6))
+
+        print("Initializing optimizer...")
+        optimizer = torch.optim.Adam(
+            params=model.parameters(), lr=self.hparams.learning_rate
+        )
+        test_batch = test_dataset.sample_test_batch(size=self.hparams.test_size)
+        for i, item in enumerate(test_batch):
+            text, mel, spk = item
+            self.log(
+                f"image_{i}/ground_truth",
+                0,
+                image=plot_tensor(mel.squeeze()),
+            )
```

Comment:
should this be included?

Reply:
It should be included but it doesn't need to be called at every evaluation. I'll put it before the training loop begins